packages/dnslink-api/src/index.js

```diff
@@ -58,9 +61,25 @@ server.get("/dnslink/:name", async (req, res) => {
 
     const skylink = matchSkylink[1];
 
+    // check if _dnslink records contain skynet-sponsor-key entries
+    const sponsors = records.filter((record) => sponsorRegExp.test(record));
+
+    if (sponsors.length > 1) {
```

Comment:
don't we need to handle the case where the length is 0

Reply:
if the length is zero, we just have a non-sponsored skylink. that's perfectly fine, if the user is using a portal that allows anonymous downloads or they are logged in in their portal.